pkg/api/vlabs/types.go

```diff
@@ -158,22 +158,32 @@ type CustomNodesDNS struct {
 	DNSServer string `json:"dnsServer,omitempty"`
 }
 
+// WindowsLicenseType represents Windows license type
+type WindowsLicenseType string
+
+const (
+	// WindowsLicenseTypeServer specifies that the image or disk that is being used was licensed server on-premises.
+	WindowsLicenseTypeServer WindowsLicenseType = "Windows_Server"
+)
+
 // WindowsProfile represents the windows parameters passed to the cluster
 type WindowsProfile struct {
-	AdminUsername          string            `json:"adminUsername,omitempty"`
-	AdminPassword          string            `json:"adminPassword,omitempty"`
-	CSIProxyURL            string            `json:"csiProxyURL,omitempty"`
-	EnableCSIProxy         *bool             `json:"enableCSIProxy,omitempty"`
-	ImageRef               *ImageReference   `json:"imageReference,omiteempty"`
-	ImageVersion           string            `json:"imageVersion,omitempty"`
-	WindowsImageSourceURL  string            `json:"WindowsImageSourceUrl"`
-	WindowsPublisher       string            `json:"WindowsPublisher"`
-	WindowsOffer           string            `json:"WindowsOffer"`
-	WindowsSku             string            `json:"WindowsSku"`
-	WindowsDockerVersion   string            `json:"windowsDockerVersion"`
-	Secrets                []KeyVaultSecrets `json:"secrets,omitempty"`
-	SSHEnabled             *bool             `json:"sshEnabled,omitempty"`
-	EnableAutomaticUpdates *bool             `json:"enableAutomaticUpdates,omitempty"`
+	AdminUsername             string            `json:"adminUsername,omitempty"`
+	AdminPassword             string            `json:"adminPassword,omitempty"`
+	CSIProxyURL               string            `json:"csiProxyURL,omitempty"`
+	EnableCSIProxy            *bool             `json:"enableCSIProxy,omitempty"`
+	ImageRef                  *ImageReference   `json:"imageReference,omitempty"`
+	ImageVersion              string            `json:"imageVersion,omitempty"`
+	WindowsImageSourceURL     string            `json:"WindowsImageSourceUrl"`
+	WindowsPublisher          string            `json:"WindowsPublisher"`
+	WindowsOffer              string            `json:"WindowsOffer"`
+	WindowsSku                string            `json:"WindowsSku"`
+	WindowsDockerVersion      string            `json:"windowsDockerVersion"`
+	Secrets                   []KeyVaultSecrets `json:"secrets,omitempty"`
+	SSHEnabled                *bool             `json:"sshEnabled,omitempty"`
+	EnableAutomaticUpdates    *bool             `json:"enableAutomaticUpdates,omitempty"`
+	IsCredentialAutoGenerated *bool             `json:"isCredentialAutoGenerated,omitempty"`
```

Comment:
no coverage for this yet?

Reply:
"IsCredentialAutoGenerated"

Reply:
Normally we do not add more test cases for this file. This will only be used in aks-engine.

Reply:
if the "IsCredentialAutoGenerated" is only used in aks-engine. then I would suggest remove this from the PR.

Reply:
no. it is for aks. When users do not provide username and password to create a k8s cluster with VMSS and azure cni, AKS will generate username and password automatically and set this to true.

Reply:
But we need to add `enableAHUB` in WindowsProfile into aks-engine. That is why you see `IsCredentialAutoGenerated ` in this PR. When I added `IsCredentialAutoGenerated `, I did not add it in this file.